Sources/BSplineLib/ParameterSpaces/parameter_space.hpp

```diff
@@ -259,15 +267,14 @@ class ParameterSpace {
   virtual Index_ FindFirstNonZeroBasisFunction(
       ParametricCoordinate_ const& parametric_coordinate,
       Tolerance const& tolerance = kEpsilon) const;
+
+  virtual KnotSpans_
+  FindKnotSpans(ParametricCoordinate_ const& parametric_coordinate,
+                Tolerance const& tolerance = kEpsilon) const;
   virtual BezierInformation_
   DetermineBezierExtractionKnots(Dimension const& dimension,
                                  Tolerance const& tolerance = kEpsilon) const;
 
-  virtual Type_
```

Comment:
Please add a comment to the EvaluateBasisFunctionPerDimension routine, so it is clear from where you took the algorithm (with citation, page and alg. ref). Also, the class documentation seems outdated

Reply:
done